src/pages/settings-general/settings-general.spec.ts

```diff
@@ -46,24 +48,54 @@ describe("SettingsGeneralPage", () => {
   it("should initialize correctly", () => {
     expect(comp.nightModeStatus).toBe(false);
     expect(comp.activeLanguage).toBeUndefined();
-  })
-
-  it("should have undefined as saved language", (done) => {
-    comp.settingsService.getLanguageSettings().then(data => {
-      expect(data).toBeUndefined();
-      done();
-    })
-  })
-
-  it("should have undefined as saved theme", (done) => {
-    comp.settingsService.getActiveTheme().subscribe(data => {
-      expect(data).toBe("light-theme");
-      done();
-    })
-  })
-
-  it("should have en as default language after setting it in the translate module", () => {
-    comp.translate.setDefaultLang("en")
-    expect(comp.translate.getDefaultLang()).toBe("en");
-  })
+    expect(comp.selectedTheme).toBeUndefined();
+  });
+
+  it("should change languages and set nightmode switch appropriately", (done) => {
```

Comment:
Testing two separate things in a unit test is generally discouraged. I would propose creating two unit tests based on this one:
1) should change languages appropriately
2) should set nightmode correctly